components/delivery-option-messages.js

```diff
@@ -0,0 +1,153 @@
+// Delivery Frequency codes
+const ONLY_WEEKEND_DELIVERY_FREQ = 'NWE';
+const FIVE_DAYS_WEEK_DELIVERY_FREQ = 'N5D';
+const SIX_DAYS_WEEK_DELIVERY_FREQ = 'N6D';
```

Comment:
This time I went through the doc carefully and I get what is going on.
The criteria are quite complex, we need to get the wording right so that we don't struggle in the future.

So these are partial matches of product codes.
We use them to derive the frequency code `(A1, A5, A6)`.

I am unsure what is the best way to write this but the current naming suggests that NWE, N5D, N6D are frequency code.
I'm concerned that people (future us) would get confused by this.

I think we can start with more detailed comment (I know it's repeated further below and should generally avoid going comment heavy, but in this instance, I'd say it's acceptable for clarity?).

```suggestion
// Partial matches on product codes NWE, N5D, N6D
// are used to derive frequency codes: 
// A1: ONLY_WEEKEND
// A5: FIVE_DAYS_WEEK
// A6: SIX_DAYS_WEEK

const ONLY_WEEKEND_DELIVERY_FREQ = 'NWE';
const FIVE_DAYS_WEEK_DELIVERY_FREQ = 'N5D'; 
const SIX_DAYS_WEEK_DELIVERY_FREQ = 'N6D';
```

I need a break and am not very inspired but hopefully it makes sense.
(@valeksiev is you have any suggestions?)

Reply:
You are right! Let's create a mapping to keep things more clear.